### DISCUSSION TITLE
Integrating Edyn with premake compilation issues.

### DISCUSSION THREAD

**Discussion by njmayes** [xissburg/edyn#100] [Q&A] [answered]
Hi there, I'm currently trying to use Edyn with my own project that uses EnTT, however I am using premake for my build system so I'm trying to recreate the configuration needed, but I'm having some compilation errors. I'm struggling to work out the cause of them, so whilst I know Edyn isn't designed to be used with premake, I'd really appreciate any help that could be given.

I'm currently trying to compile on windows x86_64 and at the moment I'm getting a template error on the `visit_component(...)` call on line 101 of `shapes.hpp`, from the `visit_shape(...)` call at line 36 of `raycast.cpp`- 
`could not deduce template argument for 'const std::tuple<entt::basic_view<entt::entity,entt::type_list<Other...>,entt::type_list<>,void>...> &' from 'const edyn::tuple_of_shape_views_t'`  

with
`'void edyn::visit_component(std::tuple<_Rest...>,IndexType,entt::entity,const std::tuple<entt::basic_view<entt::entity,entt::type_list<Other...>,entt::type_list<>,void>...> &,VisitorType)': template parameter 'Ts' is ambiguous`.

I'm also getting an error on line 306 of `server_snapshot_importer.hpp`:

`expression did not evaluate to a constant`:
`306: constexpr auto action_history_index = index_of_v<unsigned, action_history, Components...>;`

I believe I have my premake file setup with everything that is needed by comparing with CMakeLists.txt, I have all the source files being built (copied from add_library in CMakeLists.txt:28-127) and windows/unix specific time.cpp files specified. I have winmm.lib linked on windows and pthread on linux, and compiling with /bigobj on windows also.

It's very possible that there's something wrong with how I have integraded EnTT using premake as well, so if these errors indicate that let me know. I can provide more details of the errors but didn't want the question to be too long.

Edit: I can safely get rid of the second error by just using const instead of constexpr, but unsure why its not able to evaluate it at compile time.

**Comment by njmayes** [comment, marked as answer]
Referred to your comments on issue #94, and seem to have fixed this issue by reverting EnTT to the v3.10.1 tag.

**Reply by xissburg** [reply]
Did that solve the issue with `visit_component` as well?

**Reply by njmayes** [reply]
It solved the visit_component issue, but not the constexpr one, but at least I can work around that one.
